lib/spack/spack/cmd/create.py

```diff
@@ -545,7 +578,8 @@ def __call__(self, stage, url):
         ]
 
         # Peek inside the compressed file.
-        if stage.archive_file.endswith('.zip'):
+        if (stage.archive_file.endswith('.zip') or
+                '.zip#' in stage.archive_file):
```

Comment:
The archive_file gets saved with the `#sha256=...` in the filename. If there is a way to fix that on the stage.py side of things, I would prefer that.